editor/core/query_params.py

```diff
@@ -15,20 +15,40 @@ class QueryParams:
     OPEN_RECORD_SET = "recordSet"
 
 
+def _has_query_params() -> bool:
+    """Returns whether `query_params` is available, which depends on the streamlit
+    version. If not, st.experimental_get_query_params() and
+    st.experimental_set_query_params() should be used."""
+    return hasattr(st, "query_params")
+
+
 def _get_query_param(name: str) -> str | None:
     """Gets query param with the name `name`."""
-    param = st.query_params.get_all(name)
+    if _has_query_params():
+        param = st.query_params.get_all(name)
+    else:
+        param = st.experimental_get_query_params()[name]
     if isinstance(param, list) and len(param) > 0:
         return param[0]
     return None
 
 
 def _set_query_param(param: str, new_value: str) -> str | None:
-    params = st.query_params
-    if params.get_all(param) == [new_value]:
-        # The value already exists in the query params.
-        return
-    params[param] = new_value
+    """Sets query param with the name `name` to `new_value`."""
+    if _has_query_params():
+        params = st.query_params
+        if params.get_all(param) == [new_value]:
+            # The value already exists in the query params.
+            return
+        params[param] = new_value
+    else:
+        params = st.experimental_get_query_params()
+        if params.get(param) == [new_value]:
+            # The value already exists in the query params.
+            return
+        new_params = {k: v for k, v in params.items() if k != param}
```

Comment:
```
del params[param]
```

But why do you need to copy like this the dict to then reaffect the new value. Why not just `new_params[param] = new_value`?

Reply:
This is how we had it before. I think because otherwise the `set` would erase existing values.